Azure/sonic-platform-daemons
# [sfp-refactoring] xcvrd: add initial support for CMIS application initialization

Signed-off-by: Dante Su <dante.su@broadcom.com>



#### Why I did it

Add initial support for CMIS application initialization

#### How I did it

Add CmisManagerTask to handle the per-port CMIS application initialization in parallel

#### How to verify it

1. Bring up Quanta IX9 with the default configuration and CMIS 4.0/5.0 QSFPDD modules
2. Issue 'show logging xcvrd' to check if the application initialization succeeded
3. Issue 'show interfaces transceiver eeprom --dom' to verify both the static and DOM information parsers
4. Issue 'sudo sfputil show eeprom --dom' to verify both the static and DOM information parsers

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog

<table>
<tr>
<th>Repo</th>
<th>PR title</th>
<th>State</th>
</tr>
<tr>
<td>SONiC</td>
<td><a href='https://github.com/Azure/SONiC/pull/876'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/SONiC/876"></td>
</tr>
<tr>
<td>sonic-buildimage</td>
<td><a href='https://github.com/Azure/sonic-buildimage/pull/9157'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-buildimage/9157"></td>
</tr>
<tr>
<td>sonic-platform-common</td>
<td><a href='https://github.com/Azure/sonic-platform-common/pull/219'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-platform-common/219"></td>
</tr>
<tr>
<td>sonic-platform-daemon</td>
<td><a href='https://github.com/Azure/sonic-platform-daemons/pull/217'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-platform-daemons/217"></td>
</tr>
<td>sonic-utilities</td>
<td><a href='https://github.com/Azure/sonic-utilities/pull/1855'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-utilities/1855"></td>
</table>

#### A picture of a cute animal (not mandatory but encouraged)
